packages/@n8n/decorators/src/shutdown/shutdown.service.ts

```diff
@@ -3,7 +3,7 @@ import { type Class, ErrorReporter } from 'n8n-core';
 import { Logger } from 'n8n-core';
```

Comment:
This file should be moved back to `cli`. 
`@OnShutdown` should only collect metadata into a state object that this service then uses to setup the shutdown sequence.